quantum/action_util.c

```diff
@@ -398,6 +405,28 @@ void clear_weak_mods(void) {
     weak_mods = 0;
 }
 
+#ifdef SPECULATIVE_HOLD
+uint8_t get_speculative_mods(void) {
+    return speculative_mods;
+}
+
+void set_speculative_mods(uint8_t mods) {
+    speculative_mods = mods;
+}
+
+void add_speculative_mods(uint8_t mods) {
+    speculative_mods |= mods;
+}
+
+void del_speculative_mods(uint8_t mods) {
+    speculative_mods &= ~mods;
+}
+
+void clear_speculative_mods(void) {
+    speculative_mods = 0;
+}
```

Comment:
I suggest just doing these inline, unless there's something I'm missing. Breaking state by calling them from anything other than the implementation doesn't seem like it would be pretty and I don't think we can expose them.